parts/windows/csecmd.ps1

```diff
@@ -1 +1,20 @@
-"[concat('echo %DATE%,%TIME%,%COMPUTERNAME% && powershell.exe -ExecutionPolicy Unrestricted -command \"', '$arguments = ', variables('singleQuote'),'-MasterIP ',parameters('kubernetesEndpoint'),' -KubeDnsServiceIp ',parameters('kubeDnsServiceIp'),' -MasterFQDNPrefix ',variables('masterFqdnPrefix'),' -Location ',variables('location'),' -TargetEnvironment ',parameters('targetEnvironment'),' -AgentKey ',parameters('clientPrivateKey'),' -AADClientId ',variables('servicePrincipalClientId'),' -AADClientSecret ',variables('singleQuote'),variables('singleQuote'),base64(variables('servicePrincipalClientSecret')),variables('singleQuote'),variables('singleQuote'),' -NetworkAPIVersion ',variables('apiVersionNetwork'),' ',variables('singleQuote'), ' ; ', variables('windowsCustomScriptSuffix'), '\" > %SYSTEMDRIVE%\\AzureData\\CustomDataSetupScript.log 2>&1 ; exit $LASTEXITCODE')]"
\ No newline at end of file
+echo %DATE%,%TIME%,%COMPUTERNAME% && powershell.exe -ExecutionPolicy Unrestricted -command \"
+$arguments = '
+-MasterIP {{ GetKubernetesEndpoint }} 
+-KubeDnsServiceIp {{ GetParameter "kubeDNSServiceIP" }} 
+-MasterFQDNPrefix {{ GetParameter "masterEndpointDNSNamePrefix" }} 
+-Location {{ GetVariable "location" }} 
+{{if UserAssignedIDEnabled}}
+-UserAssignedClientID {{ GetVariable "userAssignedIdentityID" }} 
+{{ end }}
+-TargetEnvironment {{ GetTargetEnvironment }} 
+-AgentKey {{ GetParameter "clientPrivateKey" }} 
+-AADClientId {{ GetParameter "servicePrincipalClientId" }} 
+-AADClientSecret ''{{ GetParameter "servicePrincipalClientSecret" }}''
```

Comment:
servicePrincipalClientSecret was base64-encode in aks-engine. Here I did not.
Because:
1. The CSE command is invisible, wrapped in protectedSetting
2. We need extra code to base64-encode servicePrincipalClientSecret in our defined function or by arm template function base64.  the former one is requires extra function to maintain the encoding for [both vault and non-vault case](https://github.com/Azure/AgentBaker/blob/master/pkg/agent/params.go#L238). the latter one will be trivial to have both arm template and go-template function
3. Linux also use plain servicePrincipalClientSecret in protected CSE setting

Reply:
base64 can avoid the issue when the password contains some special characters, for example `"`.

Reply:
Resolve this conversation because we don't need this cse command file now.

Reply:
I gave up using `parts/windows/csecmd.ps1` for parsing cse command for now, but use the one copied directly from aks-engine with arm variables for safety and simplicity.
But I'll continue to pick up this piece of work to  have `parts/windows/csecmd.ps1` works afterwards.

Reply:
base64 encoding is better, maybe you can help fix linux too when you do this for windows?